sw/device/lib/dif/dif_spi_host.c

```diff
@@ -0,0 +1,388 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/dif/dif_spi_host.h"
+
+#include <assert.h>
+#include <stddef.h>
+
+#include "sw/device/lib/arch/device.h"
+#include "sw/device/lib/base/bitfield.h"
+#include "sw/device/lib/base/mmio.h"
+#include "sw/device/lib/runtime/hart.h"
+#include "sw/device/lib/runtime/log.h"
+
+#include "spi_host_regs.h"  // Generated.
+
+// We create weak symbol aliases for the FIFO write and read functions so the
+// unit tests can provide mocks.  The mocks provide for separate testing of
+// the FIFO functions and the overall transaction management functions.
+void spi_host_fifo_write_alias(const dif_spi_host_t *spi_host, const void *src,
+                               uint16_t len)
+    __attribute__((weak, alias("dif_spi_host_fifo_write")));
```

Comment:
There's a macro somewhere for `weak`. If you need alias, I think we should define `OT_SYMBOL_ALIAS` too.

That said... why do they need to be marked as aliases? Can't you put weak annotations in the header?

Reply:
I need to test the originals (`dif_spi_host_fifo_{read,write}`), so I can't make them weak and override them.  I also need to mock out the call sites during the test, so I created weak aliases.

In normal usage, the aliases will refer to the original functions.  Under test, the weak aliases will be replaced with the mock machinery for testing.